### PR TITLE
Add toml formatting

### PR DESCRIPTION
**Pull request by cycraig**
# Description of change
Adds a [`dprint`](https://github.com/dprint/dprint) configuration to format toml files, specifically Cargo.toml. This mainly reorders keys and formats whitespace to match the official conventions: https://github.com/rust-dev-tools/fmt-rfcs/blob/master/guide/cargo.md

Added a new task to the GitHub Actions `format` workflow to enforce the new style.

This also bumps the pinned nightly version in CI from `nightly-2021-07-05` to `nightly-2021-09-07` (previous version was pinned due to CI breakages).

### Usage

- Install `dprint` from source:
  ```
  cargo install dprint
  ```
  or pre-packaged:
  ```
  npm install -g dprint
  ```

- Check formatting at the root directory of "identity.rs": 
  ```
  dprint check
  ```

- Fix formatting:
  ```
  dprint fmt
  ```

## Type of change

- [x] Chore/bug fix (a non-breaking change which fixes an issue)
- [ ] Enhancement (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Fix

## Change checklist
Add an `x` to the boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/build-and-test.yml [hunk 1]
```diff
@@ -193,6 +193,7 @@ jobs:
             sccache-path: /home/runner/.cache/sccache
     env:
       SCCACHE_DIR: ${{ matrix.sccache-path }}
+      CARGO_INCREMENTAL: 0 # disabled to reduce target cache size
```
Comment (eike-hass) [inline .github/workflows/build-and-test.yml:196]: In my understanding this makes the cache smaller, but also decreases its usefulness, is that correct?

File: .github/workflows/format.yml [hunk 2]
```diff
@@ -26,21 +26,34 @@ jobs:
...
         uses: actions-rs/cargo@v1
         with:
           command: fmt
           args: --manifest-path ./libjose/Cargo.toml --all -- --check
+
+      # Use `dprint` to check Cargo.toml formatting.
+      # To fix, run `dprint fmt` locally.
+      - name: Cargo.toml fmt check
+        run:
+          dprint check
```
Comment (eike-hass) [inline .github/workflows/format.yml:59]: as an outcome the job will fail if any of the `.toml` are not formatted correctly?

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (cycraig) [reply .github/workflows/build-and-test.yml:196]: Potentially. I've seen some reports of incremental compilation slowing builds too, however I did not notice much difference locally.

The main point is that the `wasm` target cache is the largest by far at ~1173 MB, compared to ~300 MB for the others. So any even minor reduction in size would be nice.
https://github.com/iotaledger/identity.rs/runs/3568663806?check_suite_focus=true

File: .github/workflows/format.yml (hunk 2 above)
Comment (cycraig) [reply .github/workflows/format.yml:59]: Yes, it prints the formatting error and returns a non-zero exit code. E.g.

```
$dprint check
Error formatting C:\Work\iota\identity.rs\rustfmt.toml. Message: Line 8, column 16: expected value

  wrap_comments =

  true

$LASTEXITCODE
1                                            
```

File: .github/workflows/build-and-test.yml (hunk 1 above)
Comment (eike-hass) [reply .github/workflows/build-and-test.yml:196]: Oh yeah, that's totally worth it!

**Review by eike-hass — APPROVED** [review]

**Review by HenriqueNogara — APPROVED** [review]

**Pull request merged**